### PR TITLE
PR: Added Support for Binance RPC

### PR DESCRIPTION
**Pull request by weboluchi**

### CODE REVIEW

File: packages/rpc/src/binance/BinanceRPC.ts [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+    constructor(rpcUrl: string) {
+        this.rpcUrl = rpcUrl;
+    }
+
+    private query(): Query {
+        return new Query(this.rpcUrl);
+    }
+
+    async getAccount(address: string): Promise<BinanceAccount> {
+        let response = await axios.get(this.query().getAccount(address));
```
Comment (kolya182) [inline packages/rpc/src/binance/BinanceRPC.ts]:
```suggestion
        const { data } = await axios.get(this.query().getAccount(address));
```

File: packages/rpc/src/binance/BinanceRPC.ts [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+        return plainToClass(BinanceAccount, response.data);
```
Comment (kolya182) [inline packages/rpc/src/binance/BinanceRPC.ts]:
```suggestion
        return plainToClass(BinanceAccount, data);
```

**Comment by kolya182** [comment]
Committed file at `packages/rpc/src/binance/models/http` ?

File: packages/rpc/src/binance/BinanceRPC.ts (hunk 1 above)
Comment (weboluchi) [reply packages/rpc/src/binance/BinanceRPC.ts]: Working on the suggestion. Thank you

File: packages/rpc/src/binance/BinanceRPC.ts (hunk 2 above)
Comment (weboluchi) [reply packages/rpc/src/binance/BinanceRPC.ts]: Working on the suggestion. Thank you

**Comment by weboluchi** [comment]
The CI lint check throws errors in other parts of the project other than the part I worked on hence preventing the pull request from completion. 

See screenshots below : 

[image](https://user-images.githubusercontent.com/22542946/64543173-a3abc880-d31c-11e9-8afb-1e92a0647380.png)

image

image

The failing test seems to be coming from the **TrustApi.test.ts**, **utils.test.ts** of the API package.

Do I attempt to modify these parts to enable the completion of the pull request?

File: packages/rpc/src/binance/Query.ts [hunk 3]
```diff
@@ -0,0 +1,18 @@
+export class Query {
+    rpcUrl: string;
+
+    constructor(rpcUrl: string) {
+        this.rpcUrl = rpcUrl;
+    }
+
+    getNodeInfo = () => this.uri(`v1/node-info`);
+    getAccount = (address: string) => this.uri(`v1/account/${address}`);
+    getMarket = (limit?: number, offset?: number) => this.uri(`v1/markets?limit=${(limit == undefined)? 100 : limit}&offset=${(offset == undefined)? 0 : offset}`);
```
Comment (leoneparise) [inline packages/rpc/src/binance/Query.ts]: Use default values instead of checking for undefined.

**Comment by leoneparise** [comment]
Please try to merge current master to make CI work again. Another thing I would is add an `index.ts` in `models` and export only the models you think should be public. From there I would add `export * from './binance/models'` to main `index.ts` file.

File: packages/rpc/src/binance/Query.ts (hunk 3 above)
Comment (weboluchi) [reply packages/rpc/src/binance/Query.ts]: Working on that now. Thank you

**Comment by weboluchi** [comment]
> I would is add an `index.ts` in `models` and export only the models you think should be public. From there I would add `export * from './binance/models'` to main `index.ts` file.

This has been done.

File: packages/rpc/src/binance/BinanceRPC.ts [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+        const {data} = await axios.get(this.query().getAccount(address));
+        return plainToClass(BinanceAccount, data);
+    }
+
+    async getMarket(limit?: number, offset?: number): Promise<BinanceMarket> {
+        const {data} = await axios.get(this.query().getMarket(limit,offset));
+        return plainToClass(BinanceMarket, data);
+    }
+
+    async getNodeInfo(limit?: number, offset?: number): Promise<BinanceNodeInfo> {
```
Comment (dpereskokov) [inline packages/rpc/src/binance/BinanceRPC.ts]: Params seem to be redundant

**Pull request closed without merging**